File: Packets/src/main/java/me/ImSpooks/core/packets/collection/database/PacketRequestCollection.java
package me.ImSpooks.core.packets.collection.database;

import lombok.Getter;
import me.ImSpooks.core.packets.init.Packet;
import me.ImSpooks.core.packets.init.channels.WrappedInputStream;
import me.ImSpooks.core.packets.init.channels.WrappedOutputStream;

import java.io.IOException;

/**
 * Created by Nick on 09 okt. 2019.
 * Copyright © ImSpooks
 */
public class PacketRequestCollection extends Packet {

    @Getter private String collection;

    public PacketRequestCollection(String collection) {
        this.collection = collection;
    }

    public PacketRequestCollection() {
    }

    @Override
    public void send(WrappedOutputStream out) throws IOException {
        out.writeString(this.collection);
    }

    @Override
    public void receive(WrappedInputStream in) throws IOException {
        this.collection = in.readString();
    }
}

File: ClientBridge/src/main/php/client/packets/collection/database/PacketRequestCollection.php
<?php
/**
 * Created by Nick on 25 okt 2019.
 * Copyright © ImSpooks
 */

namespace client\packets\collection\database;

use client\packets\init\Packet;
use client\packets\init\channels\WrappedInputStream;
use client\packets\init\channels\WrappedOutputStream;

class PacketRequestCollection extends Packet {

    /** @var string */ 	private $collection;

    public function __construct(string $collection = "") {
        $this->collection = $collection;
    }

    public function receive(WrappedInputStream $in) {
        $this->collection = $in->readString();
    }

    public function send(WrappedOutputStream $out) {
        $out->writeString($this->collection);
    }

    public function getCollection(): string {
        return $this->collection;
    }

}
